portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,298 @@
+// require(['dotenv']).config({path: './.env'});
+
+var GoogleAuth;
+
+// Scopes for API access to Google Calendar
+const SCOPE_READ_ONLY = 'https://www.googleapis.com/auth/calendar.readonly';
+const SCOPE_READ_WRITE = 'https://www.googleapis.com/auth/calendar';
+
+// For the discovery document for Google Calendar API.
+const DISCOVERY_URL =
+    'https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest';
+
+const CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+// Constants for error codes during OAuth process.
+const ERROR_CODE_POPUP_CLOSED = 'popup_closed_by_user';
+const ERROR_CODE_ACCESS_DENIED = 'access_denied';
+
+/**
+ * Loads the API's client and auth2 modules.
+ * Calls the initClient function after the modules load.
+ */
+function initiateCalendarAuth() {
+  gapi.load('client:auth2', initClient);
+}
+
+/** Starts authentication flow based on current user's login status. */
+function initClient() {
+  // Hide the error messages by default.
+  // These messages will appear if authentication errors are caught.
+  hideErrorMessages();
+
+  // Initializes the gapi.client object, which app uses to make API requests.
+  // Initially, the scope is read-only to view user's Google Calendar.
+  gapi.client
+      .init({
+        'apiKey': API_KEY,
```

Comment:
I think we don't need the quotes in the keys.

Reference:
https://google.github.io/styleguide/jsguide.html#features-objects-mixing-keys

Ex:
.init({
  apiKey: API_KEY,
  ...

Reply:
Updated.